src/plugins/apm_oss/server/index.ts

```diff
@@ -13,6 +13,17 @@ import { APMOSSPlugin } from './plugin';
 const deprecations: ConfigDeprecationProvider = ({ unused }) => [
   unused('fleetMode'),
   unused('indexPattern'),
+  (settings, fromPath, addDeprecation) => {
+    if (settings.apm_oss?.enabled !== undefined) {
+      addDeprecation({
+        title: `Setting "${fromPath}.enabled" is deprecated`,
+        message: `"${fromPath}.enabled" has been deprecated and will be removed in 8.0.`,
+        correctiveActions: {
+          manualSteps: [`Remove ${fromPath}.enabled from your Kibana config.`],
+        },
+      });
+    }
+  },
```

Comment:
Yeah, I had considered the same thing but stopped short of doing it as it seemed like a very specific use case.

But as I think about it more, I suppose we could generalize and have something like `remove()` which indicates a setting that's going away completely in a future version, but is still being used. This would be in contrast to `unused()` which indicates that a config is simply no longer needed.

So maybe it would work something like this:
- Accepts a `version` param to indicate version in which it will be removed
- Logs a deprecation warning for that config similar to the one above, prompting users to remove by the indicated version if provided
- Doesn't modify the config at all, just exists to log the deprecation as a side effect

```ts
remove(removeKey: string, removeBy?: string, details?: Partial<DeprecatedConfigDetails>): 
ConfigDeprecation;

removeFromRoot(removeKey: string, removeBy?: string, details?: Partial<DeprecatedConfigDetails>): ConfigDeprecation;
```

Come to think of it, maybe `remove()` is a confusing name if we aren't actually modifying the config? `removeBy()`? `deprecate()`?

@pgayvallet WDYT?

Reply:
Pushed an update which adds `deprecate` and `deprecateFromRoot`, lmk what you think.

Reply:
I was gonna say, imho `deprecate` and `deprecateFromRoot` make the most sense.

Reply:
> Accepts a version param to indicate version in which it will be removed
> Logs a deprecation warning for that config similar to the one above, prompting users to remove by the indicated version if provided
> Doesn't modify the config at all, just exists to log the deprecation as a side effect

Yea, sgtm